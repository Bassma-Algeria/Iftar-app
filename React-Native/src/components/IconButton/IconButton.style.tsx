import {StyleSheet} from 'react-native';
import {COLORS} from '../../theme/Colors';

const styles = StyleSheet.create({
  container: {
    width: 50,
    height: 50,
    borderRadius: 50,
    backgroundColor: COLORS.primary,
    alignItems: 'center',
    justifyContent: 'center',
  },

  icon: {
    width: '40%',
  },
});

export {styles};
