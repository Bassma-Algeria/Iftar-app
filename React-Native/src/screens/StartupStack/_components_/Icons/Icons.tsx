import React from 'react';
import {Image} from 'react-native';
import {ICONS} from '../../../../utils/constants/Icons.ts';
interface Props {}
const Email: React.FC<Props> = () => {
  return <Image source={ICONS.email} resizeMode="contain" />;
};
const Phone: React.FC<Props> = () => {
  return <Image source={ICONS.phoneLightColor} resizeMode="contain" />;
};
const OpenEye: React.FC<Props> = () => {
  return <Image source={ICONS.openEye} resizeMode="contain" />;
};
const ClosedEye: React.FC<Props> = () => {
  return <Image source={ICONS.closedEye} resizeMode="contain" />;
};
export {Email, Phone, OpenEye, ClosedEye};
