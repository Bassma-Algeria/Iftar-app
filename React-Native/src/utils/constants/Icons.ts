const ICONS = {
  moon: require('../../assets/icons/moon.png'),
  restaurantMarker: require('../../assets/icons/restaurant-marker.png'),
  clock: require('../../assets/icons/clock.png'),
  clockBrown: require('../../assets/icons/clock-brown.png'),
  location: require('../../assets/icons/location.png'),
  phone: require('../../assets/icons/phone.png'),
  leftArrow: require('../../assets/icons/left-arrow.png'),
  email: require('../../assets/icons/email.png'),
  closedEye: require('../../assets/icons/closed-eye.png'),
  openEye: require('../../assets/icons/open-eye.png'),
  phoneLightColor: require('../../assets/icons/phone-light-color.png'),
  myLocation: require('../../assets/icons/my-location.png'),
  search: require('../../assets/icons/search.png'),
  profile: require('../../assets/icons/profile.png'),
  plus: require('../../assets/icons/plus.png'),
  locationFilled: require('../../assets/icons/location-filled.png'),
  profilePrimary: require('../../assets/icons/profile-primary.png'),
  edit: require('../../assets/icons/edit.png'),
  addPicture: require('../../assets/icons/add-picture.png'),
  done: require('../../assets/icons/done.png'),
  trash: require('../../assets/icons/trash.png'),
  rightArrow: require('../../assets/icons/right-arrow.png'),
  profileBrown: require('../../assets/icons/profile-brown.png'),
};

export {ICONS};
