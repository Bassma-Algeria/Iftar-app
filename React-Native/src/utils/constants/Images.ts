const IMAGES = {
  latterns1: require('../../assets/images/latterns-1.png'),
  latterns2: require('../../assets/images/latterns-2.png'),
};

export {IMAGES};
