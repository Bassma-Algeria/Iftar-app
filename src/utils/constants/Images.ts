const IMAGES = {
  latterns1: require('../../assets/images/latterns-1.png'),
  latterns2: require('../../assets/images/latterns-2.png'),
  noImage: require('../../assets/images/no-image.png'),
};

export {IMAGES};
