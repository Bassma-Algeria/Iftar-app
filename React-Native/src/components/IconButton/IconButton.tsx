import React from 'react';
import {
  TouchableOpacity,
  ViewStyle,
  StyleProp,
  Image,
  ImageSourcePropType,
  ImageStyle,
} from 'react-native';
import {Shadow} from 'react-native-shadow-2';

import {COLORS} from '../../theme/Colors';

import {styles} from './IconButton.style';

interface Props {
  icon: ImageSourcePropType;
  style?: StyleProp<ViewStyle>;
  size?: number;
  iconWidthPercentage?: string;
  onPress?: () => void;
}

const IconButton: React.FC<Props> = ({icon, style, onPress, size, iconWidthPercentage}) => {
  let customStyle: StyleProp<ViewStyle> = {};
  let customIconStyle: StyleProp<ImageStyle> = {};

  if (size) {
    customStyle.width = size;
    customStyle.height = size;
  }

  if (iconWidthPercentage) {
    customIconStyle.width = iconWidthPercentage;
  }

  return (
    <Shadow containerViewStyle={style} startColor={COLORS.primaryShadow} distance={2}>
      <TouchableOpacity
        style={[styles.container, customStyle]}
        onPress={onPress}
        activeOpacity={0.9}>
        <Image source={icon} style={[styles.icon, customIconStyle]} resizeMode="contain" />
      </TouchableOpacity>
    </Shadow>
  );
};

export {IconButton};
