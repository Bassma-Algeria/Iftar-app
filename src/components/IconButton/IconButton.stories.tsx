import React from 'react';
import {action} from '@storybook/addon-actions';
import {storiesOf} from '@storybook/react-native';

import {IconButton} from './IconButton';
import {ICONS} from '../../utils/constants/Icons';

const IconButtonStories = storiesOf('IconButton', module);

IconButtonStories.add('default', () => (
  <IconButton onPress={action('clicked')} icon={ICONS.profile} />
));
