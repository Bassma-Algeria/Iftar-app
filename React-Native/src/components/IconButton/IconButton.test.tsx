import {fireEvent, render} from '@testing-library/react-native';
import React from 'react';
import {TouchableOpacity} from 'react-native';
import renderer from 'react-test-renderer';

import {ICONS} from '../../utils/constants/Icons';

import {IconButton} from './IconButton';

describe('IconButton component', () => {
  const defaultProps = {icon: ICONS.profile};

  it('should render correctly', () => {
    const tree = renderer.create(<IconButton {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to change the default styles', () => {
    const tree = renderer.create(<IconButton {...defaultProps} style={{margin: 30}} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should be able to change the size of the button', () => {
    const tree = renderer.create(<IconButton size={10} {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should run the function passed in the props when pressed', () => {
    const mockFunction = jest.fn();
    const instance = render(<IconButton {...defaultProps} onPress={mockFunction} />);

    fireEvent(instance.UNSAFE_getByType(TouchableOpacity), 'onPress');

    expect(mockFunction).toBeCalledTimes(1);
  });
});
