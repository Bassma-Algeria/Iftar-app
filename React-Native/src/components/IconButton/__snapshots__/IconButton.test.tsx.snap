// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton component should be able to change the default styles 1`] = `
<View
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#B87015",
        "borderRadius": 50,
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      },
      Object {
        "margin": 30,
      },
    ]
  }
>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth={4}
      focusable={false}
      height="100%"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": -3,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="left"
            x1="1"
            x2="0"
            y1="0"
            y2="0"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "left",
            ]
          }
          height="100%"
          propList={
            Array [
              "fill",
            ]
          }
          width={3}
          x={0}
          y={-0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth={4}
      focusable={false}
      height="100%"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": "100%",
            "position": "absolute",
            "top": 0,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="right"
            x1="0"
            x2="1"
            y1="0"
            y2="0"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "right",
            ]
          }
          height="100%"
          propList={
            Array [
              "fill",
            ]
          }
          width={3}
          x={0}
          y={-0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth="100%"
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "top": "100%",
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="bottom"
            x1="0"
            x2="0"
            y1="0"
            y2="1"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "bottom",
            ]
          }
          height={3}
          propList={
            Array [
              "fill",
            ]
          }
          width="100%"
          x={-0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth="100%"
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "top": -3,
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="top"
            x1="0"
            x2="0"
            y1="1"
            y2="0"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "top",
            ]
          }
          height={3}
          propList={
            Array [
              "fill",
            ]
          }
          width="100%"
          x={-0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth={4}
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": -3,
            "position": "absolute",
            "top": -3,
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={3}
            cy={3}
            fx={3}
            fy={3}
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="topLeft"
            rx={3}
            ry={3}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M0,3 a3,3 0 0 1 3 -3 v3 a0,0 0 0 0 0,0 h-3 Z"
          fill={
            Array [
              1,
              "topLeft",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth={4}
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": "100%",
            "position": "absolute",
            "top": -3,
            "transform": Array [
              Object {
                "translateX": -0,
              },
            ],
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={0}
            cy={3}
            fx={0}
            fy={3}
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="topRight"
            rx={3}
            ry={3}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M0,0 a3,3 0 0 1 3,3 h-3 a0,0 0 0 0 0,0 v-3 Z"
          fill={
            Array [
              1,
              "topRight",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth={4}
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": -3,
            "position": "absolute",
            "top": "100%",
            "transform": Array [
              Object {
                "translateY": -0,
              },
            ],
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={3}
            cy={0}
            fx={3}
            fy={0}
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="bottomLeft"
            rx={3}
            ry={3}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M3,3 a3,3 0 0 1 -3,-3 h3 a0,0 0 0 0 0,0 v3 Z"
          fill={
            Array [
              1,
              "bottomLeft",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth={4}
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": "100%",
            "position": "absolute",
            "top": "100%",
            "transform": Array [
              Object {
                "translateX": -0,
              },
              Object {
                "translateY": -0,
              },
            ],
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={0}
            cy={0}
            fx={0}
            fy={0}
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="bottomRight"
            rx={3}
            ry={3}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M3,0 a3,3 0 0 1 -3,3 v-3 a0,0 0 0 0 0,0 h3 Z"
          fill={
            Array [
              1,
              "bottomRight",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "width": "40%",
          }
        }
      />
    </View>
  </View>
</View>
`;

exports[`IconButton component should render correctly 1`] = `
<View
  pointerEvents="box-none"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#B87015",
        "borderRadius": 50,
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      },
      undefined,
    ]
  }
>
  <View
    pointerEvents="none"
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        undefined,
      ]
    }
  >
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth={4}
      focusable={false}
      height="100%"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": -3,
            "position": "absolute",
            "top": 0,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="left"
            x1="1"
            x2="0"
            y1="0"
            y2="0"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "left",
            ]
          }
          height="100%"
          propList={
            Array [
              "fill",
            ]
          }
          width={3}
          x={0}
          y={-0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight="100%"
      bbWidth={4}
      focusable={false}
      height="100%"
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": "100%",
            "position": "absolute",
            "top": 0,
          },
          Object {
            "flex": 0,
            "height": "100%",
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="right"
            x1="0"
            x2="1"
            y1="0"
            y2="0"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "right",
            ]
          }
          height="100%"
          propList={
            Array [
              "fill",
            ]
          }
          width={3}
          x={0}
          y={-0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth="100%"
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "top": "100%",
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="bottom"
            x1="0"
            x2="0"
            y1="0"
            y2="1"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "bottom",
            ]
          }
          height={3}
          propList={
            Array [
              "fill",
            ]
          }
          width="100%"
          x={-0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth="100%"
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": 0,
            "position": "absolute",
            "top": -3,
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": "100%",
          },
        ]
      }
      width="100%"
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGLinearGradient
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={0}
            name="top"
            x1="0"
            x2="0"
            y1="1"
            y2="0"
          />
        </RNSVGDefs>
        <RNSVGRect
          fill={
            Array [
              1,
              "top",
            ]
          }
          height={3}
          propList={
            Array [
              "fill",
            ]
          }
          width="100%"
          x={-0}
          y={0}
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth={4}
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": -3,
            "position": "absolute",
            "top": -3,
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup>
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={3}
            cy={3}
            fx={3}
            fy={3}
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="topLeft"
            rx={3}
            ry={3}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M0,3 a3,3 0 0 1 3 -3 v3 a0,0 0 0 0 0,0 h-3 Z"
          fill={
            Array [
              1,
              "topLeft",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth={4}
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": "100%",
            "position": "absolute",
            "top": -3,
            "transform": Array [
              Object {
                "translateX": -0,
              },
            ],
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={0}
            cy={3}
            fx={0}
            fy={3}
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="topRight"
            rx={3}
            ry={3}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M0,0 a3,3 0 0 1 3,3 h-3 a0,0 0 0 0 0,0 v-3 Z"
          fill={
            Array [
              1,
              "topRight",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth={4}
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": -3,
            "position": "absolute",
            "top": "100%",
            "transform": Array [
              Object {
                "translateY": -0,
              },
            ],
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={3}
            cy={0}
            fx={3}
            fy={0}
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="bottomLeft"
            rx={3}
            ry={3}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M3,3 a3,3 0 0 1 -3,-3 h3 a0,0 0 0 0 0,0 v3 Z"
          fill={
            Array [
              1,
              "bottomLeft",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
    <RNSVGSvgView
      bbHeight={4}
      bbWidth={4}
      focusable={false}
      height={4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {
            "left": "100%",
            "position": "absolute",
            "top": "100%",
            "transform": Array [
              Object {
                "translateX": -0,
              },
              Object {
                "translateY": -0,
              },
            ],
          },
          Object {
            "flex": 0,
            "height": 4,
            "width": 4,
          },
        ]
      }
      width={4}
    >
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGDefs>
          <RNSVGRadialGradient
            cx={0}
            cy={0}
            fx={0}
            fy={0}
            gradient={
              Array [
                0,
                -1632211950,
                1,
                11955218,
              ]
            }
            gradientTransform={null}
            gradientUnits={1}
            name="bottomRight"
            rx={3}
            ry={3}
          />
        </RNSVGDefs>
        <RNSVGPath
          d="M3,0 a3,3 0 0 1 -3,3 v-3 a0,0 0 0 0 0,0 h3 Z"
          fill={
            Array [
              1,
              "bottomRight",
            ]
          }
          propList={
            Array [
              "fill",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "width": "40%",
          }
        }
      />
    </View>
  </View>
</View>
`;
